### PR TITLE
read license.tar.gz from repomd repositories (fate #322286)

### PR DESCRIPTION
**Pull request by wfeldt**
This includes downloading and parsing repomd.xml to find ot the real file name
and file checksum.

This reuses the existing **very** limited xml parser file_parse_xmllike() which
seems sufficient here.

### CODE REVIEW

File: auto2.c [hunk 1]
```diff
@@ -601,7 +602,10 @@ int auto2_find_repo()
   }
 
   /* get some files for lazy yast */
-  if(!err) auto2_read_repo_files(config.url.install);
+  if(!err) {
+    if(config.repomd) auto2_read_repomd_files(config.url.install);
+    auto2_read_repo_files(config.url.install);
```
Comment (jreidinger) [inline auto2.c]: so if config.repodm is set, then both reads are called? is it intention? or only read_rpomd should be called?

File: auto2.c [hunk 2]
```diff
@@ -905,6 +909,41 @@ void auto2_read_repo_files(url_t *url)
...
+  char *tmp_file = NULL;
+  static char *default_list[][2] = {
+    { "license", "/license.tar.gz" },
+  };
+
+  for(i = 0; i < sizeof default_list / sizeof *default_list; i++) {
+    // be careful not to replace an existing file unless we successfully got
+    // a new version
+
+    // first, get real file name
```
Comment (jreidinger) [inline auto2.c]: if it is only comment, I suggest to remove "first" as I try to search for second which I do not see.

File: file.c [hunk 3]
```diff
@@ -2618,3 +2616,83 @@ slist_t *file_parse_xmllike(char *name, char *tag)
...
+  if(!tag) return sl0;
+
+  if(!(f = fopen(name, "r"))) return sl0;
+
+  do {
+    buf = realloc(buf, buf_size += 0x1000);
+    i = fread(buf + buf_ptr, 1, buf_size - buf_ptr - 1, f);
+    buf_ptr += i;
+  }
+  while(buf_ptr == buf_size - 1);
```
Comment (jreidinger) [inline file.c]: is this style unified accross files? I expect to have "while" after `}` as now it looks for me like it is only `while(buf_ptr == buf_size - 1);` which is valid, just never ends. but if it is consistent with rest of code, then fine for me

File: file.c [hunk 4]
```diff
@@ -2618,3 +2616,83 @@ slist_t *file_parse_xmllike(char *name, char *tag)
...
+
+  buf[buf_ptr] = 0;
+
+  fclose(f);
+
+  if(!(buf_size = buf_ptr)) return sl0;
```
Comment (jreidinger) [inline file.c]: is `=` intetion? shouldn't it be `==` for comparison?

File: auto2.c (hunk 1 above)
Comment (wfeldt) [inline auto2.c]: Intentionally; it reads e.g. the autoinst.xml file this way.

Comment (jreidinger) [inline auto2.c]: ok, thanks for info. Can you put it also to comment in code?

File: file.c (hunk 3 above)
Comment (wfeldt) [inline file.c]: Consistent, also same for if ... else.

File: file.c (hunk 4 above)
Comment (wfeldt) [inline file.c]: Good catch.

(a) yes, '=' was intended.
(b) after the splitting of the function it's no longer needed at all

**Pull request merged**